translator/navjoy_translator.py

```diff
@@ -0,0 +1,347 @@
+
+    if not wzdx_translator.validate_wzdx(wzdx_obj, wzdx_schema):
+        logging.error(
+            'validation error more message are printed above. output file is not created because the message failed validation.')
+        return
+    with open(outputfile, 'w') as fwzdx:
+        fwzdx.write(json.dumps(wzdx_obj, indent=2))
+        print(
+            'Your wzdx message was successfully generated and is located here: ' + str(outputfile))
+
+
+# parse script command line arguments
+def parse_navjoy_arguments():
+    parser = argparse.ArgumentParser(
+        description='Translate iCone data to WZDx')
+    parser.add_argument('--version', action='version',
+                        version=f'{PROGRAM_NAME} {PROGRAM_VERSION}')
+    parser.add_argument('navjoyFile', help='navjoy file path')
+    parser.add_argument('--outputFile', required=False,
+                        default='navjoy_wzdx_translated_output_message.geojson', help='output file path')
+
+    args = parser.parse_args()
+    return args.navjoyFile, args.outputFile
+
+
+def wzdx_creator(messages, info=None, unsupported_message_callback=None):
+    if not messages:
+        return None
+   # verify info obj
+    if not info:
+        info = wzdx_translator.initialize_info(
+            '8d062f70-d53e-4029-b94e-b7fbcbde5885')
+    if not wzdx_translator.validate_info(info):
+        return None
+
+    wzd = wzdx_translator.initialize_wzdx_object(info)
+
+    for message in messages:
+        # closures can cover both directions
+        directions = message.get('data', {}).get('directionOfTraffic')
+
+        for direction in get_directions_from_string(directions):
+            # Parse closure to WZDx Feature
+            feature = parse_reduction_zone(
+                copy.deepcopy(message), direction, callback_function=unsupported_message_callback)
+            if feature:
+                wzd.get('features').append(feature)
+    if not wzd.get('features'):
+        return None
+    wzd = wzdx_translator.add_ids(wzd)
+    return wzd
+
+
+def get_linestring_index(map):
+    for i in range(len(map)):
+        if map[i].get("type") == "LineString":
+            return i
+
+
+def get_polygon_index(map):
+    for i in range(len(map)):
+        if map[i].get("type") == "Polygon":
+            return i
+
+
+def get_directions_from_string(directions_string) -> list:
+    if not directions_string or type(directions_string) != str:
+        return []
+
+    directions = []
+
+    # iterate over directions and convert short direction names to WZDx enum directions
+    directions_string = directions_string.strip()
+    for dir in directions_string.split('/'):
+        direction = STRING_DIRECTION_MAP.get(dir.lower())
+        if direction:
+            directions.append(direction)
+
+    return directions
+
+# TODO: Support additional zones per message (streetNameFrom2, ...)
+# Parse Navjoy  to WZDx
+
+
+def parse_reduction_zone(obj, direction, callback_function=None):
```

Comment:
Can we add a description for this method?

Reply:
added "Convert individual Navjoy 658 speed reduction zone to WZDx feature"